### PR TITLE
Add and use PAD character constant for ' '.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: src/main/java/org/apache/commons/lang3/StringUtils.java [hunk 1]
```diff
@@ -147,6 +147,13 @@ public class StringUtils {
      */
     public static final String SPACE = " ";
 
+    /**
+     * A space character for pad {@code ' '}.
+     *
+     * @since 3.13.0
+     */
+    public static final char PAD = ' ';
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/StringUtils.java:155]: Does not need to be public.

Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/StringUtils.java]: done

File: src/main/java/org/apache/commons/lang3/StringUtils.java [hunk 2]
```diff
@@ -147,6 +147,12 @@ public class StringUtils {
      */
     public static final String SPACE = " ";
 
+    /**
+     * A space character for pad {@code ' '}.
+     *
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/StringUtils.java:152]: No need for a blank Javadoc line.

Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/StringUtils.java]: done

File: src/main/java/org/apache/commons/lang3/StringUtils.java [hunk 3]
```diff
@@ -3808,7 +3813,7 @@ public static boolean isNumericSpace(final CharSequence cs) {
         final int sz = cs.length();
         for (int i = 0; i < sz; i++) {
             final char nowChar = cs.charAt(i);
-            if (nowChar != ' ' && !Character.isDigit(nowChar)) {
+            if (nowChar != PAD && !Character.isDigit(nowChar)) {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/StringUtils.java:3816]: The concept here is not "padding", it's "spacing", so it does not feel appropriate to use the name "PAD".  Note that we have a SPACE constant already but it's a String, not a char, so maybe we need a private space char constant. I'm not sure.

Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/StringUtils.java]: HI @garydgregory .You're right., I have the same doubt. For the moment I removed that change

**Pull request closed without merging**